java-checks/src/main/java/org/sonar/java/checks/ImmediatelyReturnedVariableCheck.java

```diff
@@ -79,12 +84,23 @@ public void visitBlock(BlockTree tree) {
             // Can only happen for non-compilable code, still, we should not report anything.
             return;
           }
-          context.reportIssue(
-            this, initializer, "Immediately " + lastTypeForMessage + " this expression instead of assigning it to the temporary variable \"" + identifier + "\".");
+          ((InternalJavaIssueBuilder) ((DefaultJavaFileScannerContext) context).newIssue())
+            .forRule(this)
+            .onTree(initializer)
+            .withMessage(String.format("Immediately %s this expression instead of assigning it to the temporary variable \"%s\".", lastTypeForMessage, identifier))
```

Comment:
`withMessage()` has already a built-in support of message format, just write
```java
.withMessage("Immediately %s this expression instead of assigning it to the temporary variable \"%s\".", lastTypeForMessage, identifier)
```